src/lib.rs

```diff
@@ -395,9 +450,16 @@ mod tests {
         )
         .unwrap();
 
+        let layer_index_to_restart = 3;
+        let mut current_layer_index = 0;
+        let mut layer_to_restart = Layer::default();
         let roots = sealer
             .map(|r| {
                 let l = r.unwrap();
+                if current_layer_index == layer_index_to_restart {
+                    layer_to_restart = l.base;
+                }
```

Comment:
Maybe it would be cleaner to collect all layers in a `layers : Vec<Layer>` vector before collecting the roots and get `layer_to_restart` through `layers[layer_index_to_restart`

Reply:
Good call, done.